integration/spark-datasource/src/main/scala/org/apache/spark/sql/carbondata/execution/datasources/SparkCarbonFileFormat.scala

```diff
@@ -143,6 +145,13 @@ class SparkCarbonFileFormat extends FileFormat
     conf.set(CarbonSQLHadoopMapReduceCommitProtocol.COMMIT_PROTOCOL, "true")
 
     new OutputWriterFactory {
+
+      /**
+       * counter used for generating unique task numbers.
+       */
+      val counter = new AtomicLong()
```

Comment:
Can we use UUID instead of this shared variable to generate the id?

Reply:
@jackylk  : ok I will check.
@ravipesala : I have used the similar logic used in SparkCarbonTableFormat, UUID is not used there because of any problem ?

Reply:
@jackylk in carbon taskid is treated as number, not a string so the queries might fail. And to change the number to the string may need more changes. So here uuid generated based on the taskid and counter.

Reply:
understood.